packages/proto-rpc/src/ClientTransport.ts

```diff
@@ -0,0 +1,138 @@
+import {
+    ClientStreamingCall,
+    ServerStreamingCall,
+    DuplexStreamingCall,
+    Deferred,
+    RpcTransport,
+    MethodInfo,
+    RpcError,
+    RpcOptions,
+    RpcMetadata,
+    RpcStatus,
+    UnaryCall,
+    mergeRpcOptions
+} from '@protobuf-ts/runtime-rpc'
+import { v4 } from 'uuid'
+import { TODO } from './types'
+import { RpcMessage } from './proto/ProtoRpc'
+import EventEmitter = require('events')
+import { Logger } from './Logger'
+
+export enum Event {
+    RPC_REQUEST = 'streamr:dht-transport:request-new'
+}
+
+export interface ClientTransport {
+    on(event: Event.RPC_REQUEST, listener: (deferredPromises: DeferredPromises, rpcMessage: RpcMessage, options: ProtoRpcOptions) => void): this
+}
+
+export interface DeferredPromises {
+    header: Deferred<RpcMetadata>,
+    message: Deferred<object>,
+    status: Deferred<RpcStatus>,
+    trailer: Deferred<RpcMetadata>,
+    messageParser: (bytes: Uint8Array) => object
+}
+
+export interface ProtoRpcOptions extends RpcOptions {
+    notification?: boolean
+}
+
+const logger = new Logger(module)
+
+export class ClientTransport extends EventEmitter implements RpcTransport {
+    static objectCount = 0
+    private objectId = 1
+    private stopped = false
+    protected readonly defaultOptions: TODO
```

Comment:
```suggestion
    private static objectCount = 0
    private readonly objectId: number
    private readonly defaultOptions: ProtoRpcOptions
    private stopped = false
```

Adding constraints to aid in reasoning and general cleanup.